### PR TITLE
Refactor Breathing effect with runner

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

Simplify effect with an existing runner

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/rgb_matrix/animations/breathing_anim.h [hunk 1]
```diff
@@ -2,18 +2,14 @@
 RGB_MATRIX_EFFECT(BREATHING)
 #    ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS
 
-bool BREATHING(effect_params_t* params) {
-    RGB_MATRIX_USE_LIMITS(led_min, led_max);
+hsv_t BREATHING_math(hsv_t hsv, uint8_t i, uint8_t time) {
+    time  = scale16by8(g_rgb_timer, rgb_matrix_config.speed / 8);
```
Comment (drashna) [inline quantum/rgb_matrix/animations/breathing_anim.h:6]: Not a huge fan of calling `scale16by8` twice, but the impact should be small, I think. Short of rewriting the runner itself,

**Review by drashna — APPROVED** [review]

File: quantum/rgb_matrix/animations/breathing_anim.h (hunk 1 above)
Comment (filterpaper) [reply quantum/rgb_matrix/animations/breathing_anim.h:6]: A simpler runner can be added without a time scale for effects that uses its own. Should this (and the other runner PR changes) wait for it?

Comment (drashna) [reply quantum/rgb_matrix/animations/breathing_anim.h:6]: That would probably be best. But I would say get these in first and then add/update the runners later.

Comment (filterpaper) [reply quantum/rgb_matrix/animations/breathing_anim.h:6]: Thanks, I can work on that in the next cycle.

Comment (filterpaper) [reply quantum/rgb_matrix/animations/breathing_anim.h]: > Not a huge fan of calling `scale16by8` twice, but the impact should be small, I think. Short of rewriting the runner itself,

Using `time / 2` from runner should scale it down accordingly to the original.

**Review by fauxpark — COMMENTED** [review]
Should also do this for the LED Matrix animations, if they haven't already.

**Comment by filterpaper** [comment]
> Should also do this for the LED Matrix animations, if they haven't already.

There's just one candidate—added in a separate PR.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**